Author: Javi Merino <vicho@debian.org>
Description: Don't rm /usr/bin/python when running the testsuite
 mercurial tries to delete /usr/bin/python when running the testsuite
 with "make tests TESTFLAGS=--with-hg=/usr/bin/hg".  If you're doing
 it as a user, the testsuite fails because it can't remove it.
 .
 This fixes it at least in Debian.  I'm unsure about other systems.
Bug: http://bz.selenic.com/show_bug.cgi?id=4045

--- mercurial.orig/tests/run-tests.py
+++ mercurial/tests/run-tests.py
@@ -3021,7 +3021,7 @@ class TestRunner(object):
                  sysexecutable)
             mypython = os.path.join(self._tmpbindir, pyexename)
             try:
-                if os.readlink(mypython) == sysexecutable:
+                if (mypython == sysexecutable) or (os.readlink(mypython) == sysexecutable):
                     return
                 os.unlink(mypython)
             except OSError as err:
